File: Data Structures & Algorithms/Algorithm Toolbox/week3_hash_tables/6_matching_with_mismatches/matching_with_mismatches.java
import java.util.*;
import java.util.stream.Collectors;
import java.io.*;

public class matching_with_mismatches {
    public List<Integer> solve(int k, String text, String pattern) {
        ArrayList<Integer> pos = new ArrayList<>();
        return pos;
    }

    public void run() {
        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
        PrintWriter out = new PrintWriter(System.out);
        in.lines().forEach(line -> {
            StringTokenizer tok = new StringTokenizer(line);
            int k = Integer.valueOf(tok.nextToken());
            String s = tok.nextToken();
            String t = tok.nextToken();
            List<Integer> ans = solve(k, s, t);
            out.format("%d ", ans.size());
            out.println(ans.stream()
                    .map(n -> String.valueOf(n))
                    .collect(Collectors.joining(" "))
            );
        });
        out.close();
    }

    static public void main(String[] args) {
        new matching_with_mismatches().run();
    }
}

File: Data Structures & Algorithms/Algorithm Toolbox/week3_hash_tables/6_matching_with_mismatches/matching_with_mismatches.kt
fun solve(k: Int, text: String, pattern: String): List<Int>  {
    val pos = ArrayList<Int>()
    return pos
}

fun main(args: Array<String>) {
    val inp = System.`in`.bufferedReader()
    val out = System.`out`.bufferedWriter()
    for (line in inp.readLines()) {
        val (k, s, t) = line.split(" ")
        val ans = solve(k.toInt(), s, t)
        out.write("${ans.size} ${ans.joinToString(" ")} \n");
    }
    out.close()
}
